eventsourcing/src/main/java/org/axonframework/eventsourcing/MultiEventStateApplier.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.annotation.PriorityAnnotationComparator;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.common.infra.DescribableComponent;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Implementation of {@link EventStateApplier} that applies state changes through a list of {@link EventStateApplier}
+ * instances. Every instance is called during the application of an event, regardless of whether the event was already
+ * handled by another. During construction, the list of {@link EventStateApplier} instances is sorted based on the
+ * {@link org.axonframework.common.Priority} annotation on the classes of the instances, if present.
+ *
+ * @param <M> The model type to apply the event state to.
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class MultiEventStateApplier<M> implements EventStateApplier<M>, DescribableComponent {
```

Comment:
Just a proposition, because I believe it's the `Composite` pattern. 
```suggestion
public class ComposedEventStateApplier<M> implements EventStateApplier<M>, DescribableComponent {
```

Reply:
Everywhere in the framework, so far, we use Multi-. 
That's why I kept going with Multi here as well :)